storage/command_queue.go

```diff
@@ -74,6 +75,34 @@ func (cq *CommandQueue) onEvicted(key, value interface{}) {
 	}
 }
 
+// GetWaitBatch is like GetWait, but it makes sure that all the spans are added
+// to the command queue simultaneously. That is, one can add overlapping
+// commands without them blocking on each other (in which case the command
+// would be deadlocked).
+// TODO(tschottdorf): update comment and replace GetWait with this.
+func (cq *CommandQueue) GetWaitBatch(spans []keys.Span, readOnly bool, wg *sync.WaitGroup) {
+	var pending []*[]*sync.WaitGroup
```

Comment:
this is pretty unusual. can't you write `var pending [][]*sync.WaitGroup` and then use plain-old-indexing on L101?